dask/array/routines.py

```diff
@@ -228,10 +228,12 @@ def _tensordot(a, b, axes):
     else:
         x = tensordot(a, b, axes=axes)
 
-    ind = [slice(None, None)] * x.ndim
-    for a in sorted(axes[0]):
-        ind.insert(a, None)
-    x = x[tuple(ind)]
+    if len(axes[0]) != 1:
```

Comment:
Are changes in this file a general Dask issue/limitation or are these specific for CuPy and SciPy? Dask can also use [PyData's Sparse](https://sparse.pydata.org/en/stable/), so perhaps adding a test to cover these changes for CuPy in https://github.com/dask/dask/blob/master/dask/array/tests/test_cupy.py and PyData Sparse in https://github.com/dask/dask/blob/master/dask/array/tests/test_sparse.py would be good.

Reply:
This change affects not only CuPy and SciPy but also others, so I will add tests to check that.

Reply:
I don't think we need to add a test for PyData Sparse, as there already seems to be a test for tensordot.
https://github.com/dask/dask/blob/bf17b72251ade963df679d842c9b691c52a89d8a/dask/array/tests/test_sparse.py#L97-L111